yb-voyager/cmd/importData.go

```diff
@@ -935,34 +935,65 @@ func executeSqlFile(file string) {
 	}
 
 	var errOccured = 0
-	sqlStrArray := createSqlStrArray(file, "")
-	for _, sqlStr := range sqlStrArray {
-		log.Infof("Run query %q on target %q", sqlStr[1], target.Host)
-		_, err := conn.Exec(context.Background(), sqlStr[0])
+	sqlInfoArr := createSqlStrInfoArray(file, objType)
+	for _, sqlInfo := range sqlInfoArr {
+		err := executeSqlStmtWithRetries(conn, sqlInfo, objType)
 		if err != nil {
-			log.Errorf("Run query %q on target %q: %s", sqlStr[1], target.Host, err)
-			if strings.Contains(err.Error(), "already exists") {
+			errOccured = 1
+		}
+	}
+
+	utils.WaitChannel <- errOccured
+	<-utils.WaitChannel
+}
+
+func executeSqlStmtWithRetries(conn *pgx.Conn, sqlInfo sqlInfo, objType string) error {
+	var err, err2 error
+	log.Infof("Run query %q on target %q", sqlInfo.formattedStmtStr, target.Host)
+	for retryCount := 0; retryCount <= DDL_MAX_RETRY_COUNT; {
+		_, err = conn.Exec(context.Background(), sqlInfo.stmt)
+		if err != nil {
+			log.Errorf("DDL Execution Failed %s", err)
+			if strings.Contains(strings.ToLower(err.Error()), strings.ToLower(SCHEMA_VERSION_MISMATCH_ERR)) &&
+				objType == "INDEX" {
+				// creating fresh connection
+				conn, err2 = pgx.Connect(context.Background(), target.GetConnectionUri())
```

Comment:
Note that, setting a `conn` variable here is not going to have any effect of the `conn` in the `executeSqlFile()`.

Reply:
done